vm/src/builtins/complex.rs

```diff
@@ -305,10 +305,26 @@ impl PyComplex {
     #[pymethod(magic)]
     fn repr(&self) -> String {
         let Complex64 { re, im } = self.value;
-        if re == 0.0 {
+        let re_part = if re.is_nan() {
```

Comment:
So, here's a another idea. We can re-use float formatting to get the job done (and get long float values for the `real` and `imag` parts to print out in scientific notation).

Inside `float_ops` we have `to_string` that takes an `f64` and returns it nicely formatted (it also includes handling of the `NaN`). The main issue is the sign we add, if its a `nan` for `imag` we must always prefix with a `"+"` (granted that a `real` part exists).

Reply:
That's a really great suggestion, sorry I did not check `float_ops` before 😞 
Changing this right now.